draft-ietf-oauth-selective-disclosure-jwt.md

```diff
@@ -537,447 +539,170 @@ Note: Examples in this document use `cnf` Claim defined in [@RFC7800] to include
+For presentation to a Verifier, the Holder MUST perform the following (or equivalent) steps:
 
-The Holder SHOULD verify the binding between SD-JWT and II-Disclosures Object by performing the following steps:
- 1. Check that all the claims in the II-Disclosures Object are present in the SD-JWT and that there are no claims in the SD-JWT that are not in the II-Disclosures Object
- 2. Check that the digests of the claims in the II-Disclosures Object match those in the SD-JWT
+ 1. Decide which Disclosures to release to the Verifier, obtaining proper End-User consent if necessary.
+ 2. If Holder Binding is required, create a Holder Binding JWT.
+ 3. Create the Combined Format for Presentation, including the selected Disclosures and, if applicable, the Holder Binding JWT.
+ 4. Send the Presentation to the Verifier.
 
-## Verification by the Verifier when Receiving SD-JWT and Holder-Selected Disclosures JWT {#verifier-verification}
+## Verification by the Verifier  {#verifier_verification}
 
-Verifiers MUST follow [@RFC8725] for checking the SD-JWT and, if signed, the
-HS-Disclosures JWT.
+Upon receiving a Presentation, Verifiers MUST ensure that
 
-Verifiers MUST go through (at least) the following steps before
-trusting/using any of the contents of an SD-JWT:
+ * the SD-JWT is valid, i.e., it is signed by the Issuer and the signature is valid,
+ * all Disclosures are correct, i.e., their digests are referenced in the SD-JWT, and
+ * if Holder Binding is required, the Holder Binding JWT is signed by the Holder and valid.
+
+To this end, Verifiers MUST follow the following steps (or equivalent):
 
  1. Determine if Holder Binding is to be checked according to the Verifier's policy
     for the use case at hand. This decision MUST NOT be based on whether
-    the HS-Disclosures
-    JWT is signed or not. Refer to (#holder_binding_security) for
+    a Holder Binding JWT is provided by the Holder or not. Refer to (#holder_binding_security) for
     details.
- 2. Check that the presentation consists of six period-separated (`.`) elements; if Holder Binding is not required, the last element can be empty.
- 3. Separate the SD-JWT from the HS-Disclosures JWT.
- 4. Validate the SD-JWT:
-    1. Ensure that a signing algorithm was used that was deemed secure for the application. Refer to [@RFC8725], Sections 3.1 and 3.2 for details. `none` MUST NOT be accepted.
+ 2. Separate the Presentation into the SD-JWT, the Disclosures (if any), and the Holder Binding JWT (if provided).
+ 3. Validate the SD-JWT:
+    1. Ensure that a signing algorithm was used that was deemed secure for the application. Refer to [@RFC8725], Sections 3.1 and 3.2 for details. The `none` algorithm MUST NOT be accepted.
     2. Validate the signature over the SD-JWT.
     3. Validate the Issuer of the SD-JWT and that the signing key belongs to this Issuer.
     4. Check that the SD-JWT is valid using `nbf`, `iat`, and `exp` claims, if provided in the SD-JWT.
```

Comment:
if these claims are selectively disclosed, would not be in this step, but don't know how relevant that is..

Reply:
```suggestion
    4. Check that the SD-JWT is valid using `nbf`, `iat`, and `exp` claims, if provided in the SD-JWT, and not selectively disclosed.
```